### ISSUE TITLE
Add hexadecimal, octal, and binary int literals

### ISSUE THREAD

**Issue by jakbyte** [mun-lang/mun#119]
This will add support for hexadecimal, octal, and binary notation using the prefixes `0x`, `0o`, and `0b`.
Like Rust, you should be able to use underscore spacing (as in #120) like `0xFF_FF`, and it should be case-insensitive like `0xDeAdBeEf`.

**Comment by jakbyte** [comment]
This is already partially implemented [here](https://github.com/mun-lang/mun/blob/d9b42cdffe7b474c208f465a5ea4b3bbd1e0444a/crates/mun_syntax/src/parsing/lexer/numbers.rs#L5) with a test [here](https://github.com/mun-lang/mun/blob/d9b42cdffe7b474c208f465a5ea4b3bbd1e0444a/crates/mun_syntax/src/tests/lexer.rs#L27).
It seems to be lexed properly but not parsed.

Also, the compiler panics when trying to use these:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseIntError { kind: InvalidDigit }', mun/crates/mun_hir/src/expr.rs:521:38
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by baszalmstra** [comment]
Fixed in #122
